### ISSUE TITLE
Clean up documentations

### ISSUE THREAD

**Issue by kirichoi** [sys-bio/roadrunner#720]
Roadrunner readthedocs has been restored but I see a lot of incomplete/missing pages...

**Comment by CiaranWelsh** [comment]
The C API docs for example used to have things like: 

```
    int r = rrc::getNumberOfReactions(rrHandle);
    int m = rrc::getNumberOfFloatingSpecies(rrHandle);
    int b = rrc::getNumberOfBoundarySpecies(rrHandle);
    int p = rrc::getNumberOfGlobalParameters(rrHandle);
    int c = rrc::getNumberOfCompartments(rrHandle);
    this->result = rrc::simulateEx (...)
```

**Comment by CiaranWelsh** [comment]
There used to be two separate documentation systems, doxygen and sphinx for the rr docs. There actually still is, but while revamping the rr buildsystem I consolidated the content (using [this](https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/) tutorial and [breathe](https://breathe.readthedocs.io/en/latest/)) so that we only have documentation in one place.  All content was and still is derived directly from the doc strings, from all languages. 

The options from Doxygen/Breathe/Sphinx can still be fine tuned. There are options for things like not including undocumented methods or not including private methods in the docs. I went for the most permissive option available so that everything was included.

However, one mistake I made was a confusion regarding the `src/C` folder and the `wrappers/C` folder. In short, the `src/C` folder has been used to document the C API where actually it should have been `wrappers/C`.
